client/app/version.go

```diff
@@ -0,0 +1,46 @@
+// This code is available on the terms of the project LICENSE.md file,
+// also available online at https://blueoakcouncil.org/license/1.0.0.
+
+package app
+
+import (
+	_ "embed"
+
+	"decred.org/dcrdex/dex/version"
+)
+
+/*
+Note for maintainers:
+
+The expected process for setting the version in releases is as follows:
+- Create a release branch of the form 'release-vMAJOR.MINOR'
+- Modify the Version variable below on that branch to:
+	- Remove the pre-release portion
+	- Set the build metadata to 'release.local'
+	- Example: 'Version = "0.5.0+release.local"'
+- Update the VERSION file on the master branch to the next expected version
+	while retaining a pre-release of 'pre'
+
+These steps ensure that building from source produces versions that are
+distinct from reproducible builds that override the Version via linker
+flags.
+
+Version is the application version per the semantic versioning 2.0.0 spec
+(https://semver.org/).
+
+It MUST be a full semantic version per the semantic versioning spec or
+the package will panic at runtime.  Of particular note is the pre-release
+and build metadata portions MUST only contain characters from
+semanticAlphabet.
+NOTE: The Version string is overridden on init.
+
+The version is set in a text file and go:embed-ed, so that packaging scripts can
+easily read it and stay in sync.
```

Comment:
Packaging scripts like the decred/release repo or the pkg.sh script set this with ldflags.  I don't know how that's going to work with embed, but it's probably going to break it.  The idea described in the "for maintainers" comment above is to have builds that are made with local `go build` have "+release.local" (or "-pre" if not on a release branch) while official builds made with the packaging scripts will use the overriding string, which does _not_ have that ".local" suffix.

But the client/cmd/dexc-desktop/debian-package.sh in https://github.com/decred/dcrdex/pull/2308 is setting it via ldflags so I don't see the purpose of also doing go:embed.